### DISCUSSION TITLE
different behaviour of colors when set by name i.e. blue,green,etc or by hex code, #000000 or #ffffff

### DISCUSSION THREAD

**Discussion by justvipul** [starship/starship#3092] [Q&A] [answered]
i use xresources to set colors for my terminal. i use urxvt. i have expermiented with starship by setting colors by the color name and by the actual hex codes. when i set the colors by the name then the colors change when i change the urxvt colorscheme. which is the desired effect. if the hex code is used instead then the terminal colorscheme is not respected. 
now my problem is i want the starship colors to be changed when the terminal colorscheme changes, this works fine when they are set as words, but the words colors have a very limited color sets. so what can be the efficient solution for this problem? |
desired result- a large set of colors to be possible to be chosen from, which are based on the terminal colorscheme and changes accordingly.

**Comment by davidkna** [comment]
Hex code colors are directly interpreted by your terminal as RGB values and aren't intended to be touched by any terminal themes.

For more options regarding colors that are theme-able and which don't have names in starship, you can use one of the [256 ANSI color codes](https://i.stack.imgur.com/KTSQa.png) directly by number (for example `fg:27`).

**Reply by justvipul** [reply]
unfortunately its not working. i have set fg and bg values from a range of 1 to 256, in this format, (bg:4 fg:240), but when i change my colorscheme the starship prompt does not reflect that.
>not working
[starship.toml](https://ezup.dev/p/52js4r)
[video](https://0x0.st/-YNH.mp4)
>working
[starship.toml](https://ezup.dev/p/06d45i)
[video](https://0x0.st/-YNX.mp4)

**Comment by justvipul** [comment, marked as answer]
it works only for values ranging from 1-16. anything beyond is neglected.

**Reply by justvipul** [reply]
nothing can be done for that as of now. so theres no point in pursuing this further.
